### PR TITLE
Indent JSON body while exporting it as code

### LINKED ISSUE #968: Indent JSON data while exporting it as Python code
I was testing out a web API and used the "Export flow as Python code" feature for the first time as user, and noticed an improvement.

Currently we just export the `flow.request.body` as is (independent of it's content type) but mitmproxy's interface is smart and renders different bodies differently (for eg. it indents JSON)

I think we could add this indent behaviour while exporting things as code too.

### PR DESCRIPTION
**Pull request by dufferzafar**
Fixes #968

### CODE REVIEW

File: mitmproxy/flow_export.py [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+        return data
+
+    cv = contentviews.get_content_view(
+        contentviews.get("Auto"),
+        data,
+        headers=headers,
+    )
+
+    if cv[0] == "JSON":
+        return "\n".join(l[0][1] for l in cv[1])
```
Comment (dufferzafar) [inline mitmproxy/flow_export.py]: Currently, I am just hardcoding this for JSON only as I couldn't figure out a way to make it work for all the cases. Help?

**Comment by mhils** [comment]
The problem with the content views is that they often do more than just adjust whitespace, e.g., decode protobufs. Maybe we just want to consider the JSON case here and use `requests.post(url, json=payload)` ?

**Comment by dufferzafar** [comment]
Well to be honest, I'd be happy with only the JSON case too, but do we really need to use the `json` parameter in `requests.post` ? Since we already have valid headers etc. we could make do with the `data` param too, right?

**Comment by mhils** [comment]
My understanding is that you can pass a dict/list/json thing into `json=`, whereas you pass a string otherwise?

**Comment by dufferzafar** [comment]
@mhils I guess I wasn't quite familiar with the `json` parameter, I 've updated the code to use it now!

**Comment by mhils** [comment]
Hm.. Maybe we just want to check the content type and then use `json.dumps` here? This looks like a weird "let's make a content view and then unfiddle the content view" dance here. The less cross-dependencies we have, the better :smiley:

**Comment by dufferzafar** [comment]
@mhils I've updated the code to use `json.dumps` and have also written a test. Please review :)

File: mitmproxy/flow_export.py [hunk 2]
```diff
@@ -53,8 +56,16 @@ def python_code(flow):
 
     data = ""
     if flow.request.body:
-        data = "\ndata = '''%s'''\n" % flow.request.body
-        args += "\n    data=data,"
+        json_obj = is_json(flow.request.headers, flow.request.body)
+        if json_obj:
+            # Without the separators field json.dumps() produces
+            # trailing white spaces: https://bugs.python.org/issue16333
```
Comment (Kriechi) [inline mitmproxy/flow_export.py]: Is this still a thing? As far I see it from quickly browsing the bug tracker, this should be already be fixed a long time ago...?

Comment (dufferzafar) [inline mitmproxy/flow_export.py]: The 'fix' for that issue was to update the documentation that suggests using `separators=(',', ': '))` - https://hg.python.org/cpython/rev/78bad589f205

**Comment by Kriechi** [comment]
Thanks! :tada:

**Pull request merged**